apps/ledger-live-desktop/src/newArch/features/WalletSync/hooks/useRemoveMember.ts

```diff
@@ -51,20 +51,21 @@ export function useRemoveMember({ device, member }: Props) {
   const removeMember = useCallback(
     async (member: TrustchainMember) => {
       try {
-        await runWithDevice(deviceRef.current?.deviceId, async transport => {
-          const newTrustchain = await sdkRef.current.removeMember(
-            transport,
-            trustchainRef.current as Trustchain,
-            memberCredentialsRef.current as MemberCredentials,
-            member,
-            {
-              onStartRequestUserInteraction: () => setUserDeviceInteraction(true),
-              onEndRequestUserInteraction: () => setUserDeviceInteraction(false),
-            },
-          );
+        if (!deviceRef.current) {
+          throw new Error("Device not found");
+        }
+        const newTrustchain = await sdkRef.current.removeMember(
+          deviceRef.current?.deviceId,
+          trustchainRef.current as Trustchain,
+          memberCredentialsRef.current as MemberCredentials,
```

Comment:
we shouldn't have to cast these, instead we may want to have the same trick you did with the `if(!deviceRef.current) throw`
you can have a catch all `if (!a || !b || !c) throw new Error("unexpected missing args")` for instance

also you can do `device.current.deviceId` as you just have proven to TS that it's settled. if TS can't manage it, then it"s preferable to do:

```
const device = deviceRef.current
if (!device) ...

removeMember(device.deviceId, 
```

> Side question but does it mean removeMember accepts a falsy deviceId? it probably shouldn't 🤔

Reply:
using React ref usually is done when we want to prevent an effect to be retriggered over and over, and when we instead just need to access the "current" state and not to "trigger each time it changes"